python/cudf/cudf/core/udf/utils.py

```diff
@@ -306,6 +253,16 @@ def _compile_or_get(
     we then obtain the return type from that separate compilation and
     use it to allocate an output column of the right dtype.
     """
+    # runtime check for CEC mode which is disabled for CUDA 12 for now
```

Comment:
Is checking here sufficient to catch an attempt to use CEC in enough cases? For example, indexing a series will result in a kernel launch that will elide this check and fail with the PTX version error. Running:

```
NUMBA_CUDA_LOG_LEVEL=DEBUG python -c "import cudf; cudf.Series([1, 2, 3])[0]"
```

gives output that includes:

```
== CUDA [1913] DEBUG -- call driver api: cuLaunchKernel
```

Reply:
@gmarkall this is happening because various uses of `iloc` within cuDF leverage a numba kernel [here](https://github.com/rapidsai/cudf/blob/branch-23.06/python/cudf/cudf/utils/cudautils.py#L84). 

It didn't occur to me until now that numba may need to be in CEC mode to launch kernels at all if the driver is from cuda 12.0 and the ctk is version 12.1, I assume this is because it's attempting to link PTX generated by the version of NVRTC shipped with the 12.1 ctk.

I'm a little worried this could be a bigger issue because not having working indexing more or less means that the library is broken entirely in the 12.0/12.1 configuration. 

cc @shwina @vyasr

Reply:
Yes, it does need to be in CEC mode to launch kernels generated with a 12.1 toolkit on a 12.0 driver at all. But I think in past discussions we've concluded that we're not supporting CEC at all for CUDA 12 in 23.06. My concern here was that we'll end up giving the user the slightly cryptic message about an unsupported PTX version in this case, rather than the helpful message you've added below in this function.

Reply:
If that's the case then I would imagine we need a little bit more then a runtime check here, we really need cuDF to throw a CEC related error whenever theres _any_ kind of attempt at usage beyond just importing the library.

Reply:
Is the comment here now outdated / vestigial?

Reply:
Yes. To summarize we moved towards warning on import of cuDF that it will likely be fragile rather than throwing when calling each and every API that could fail. I'll remove this comment.